core/src/test/scala/org/apache/spark/ui/UISeleniumSuite.scala

```diff
@@ -725,6 +729,30 @@ class UISeleniumSuite extends SparkFunSuite with WebBrowser with Matchers with B
     }
   }
 
+  test("Staleness of Spark UI should not last minutes or hours") {
+    withSpark(newSparkContext(
+      master = "local[2]",
+      // Set a small heart beat interval to make the test fast
+      additionalConfs = Map(
+        EXECUTOR_HEARTBEAT_INTERVAL.key -> "10ms",
+        LIVE_ENTITY_UPDATE_STALENESS_LIMIT.key -> "10ms"))) { sc =>
+      val f = sc.parallelize(1 to 1000, 1000).foreachAsync { _ =>
+        // Make the task never finish so there won't be any task start/end events after the first 2
+        // tasks start.
+        Thread.sleep(300000)
```

Comment:
Nit: what about a less than 5 minutes sleep here something comparable with the eventually, like:
```
        Thread.sleep(20.seconds.toMillis)
```

Reply:
I turned on `SPARK_JOB_INTERRUPT_ON_CANCEL`, so it's not needed to change the sleep time.

Reply:
I have checked and the thread leaks are gone.